port/unix/omrsysinfo.c

```diff
@@ -3568,18 +3585,28 @@ omrsysinfo_startup(struct OMRPortLibrary *portLibrary)
 	 */
 	if (0 == attachedPortLibraries) {
 		if (omrthread_monitor_init_with_name(&cgroupEntryListMonitor, 0, "cgroup entry list monitor")) {
-			return OMRPORT_ERROR_STARTUP_SYSINFO_MONITOR_INIT;
+			rc = OMRPORT_ERROR_STARTUP_SYSINFO_MONITOR_INIT;
+			goto _end;
 		}
 	}
 	attachedPortLibraries += 1;
-	isRunningInContainer(portLibrary, &PPG_isRunningInContainer);
-#endif /* defined(LINUX) */
-	return 0;
 
-#if (defined(S390) || defined(J9ZOS390))
-	PPG_stfleCache.lastDoubleWord = -1;
-#endif
+	if (isCgroupV1Available(portLibrary)) {
+		PPG_sysinfoControlFlags |= OMR_CGROUP_V1_AVAILABLE;
+	} else if (isCgroupV2Available()) {
+		PPG_sysinfoControlFlags |= OMR_CGROUP_V2_AVAILABLE;
+	}
+	if (0 == isRunningInContainer(portLibrary, &runningInContainer)) {
```

Comment:
There should be a comment highlighting the order of functions. `isCgroupV1Available` should be called before invoking `isRunningInContainer`. Changing the order will cause `isRunningInContainer` to malfunction because it relies upon the `OMRPORT_SYSINFO_CGROUP_V1_AVAILABLE` bit which is set by `isCgroupV1Available`.